src/server.c

```diff
@@ -1691,7 +1691,7 @@ void databasesCron(void) {
     /* Expire keys by random sampling. Not required for slaves
      * as master will synthesize DELs for us. */
     if (server.active_expire_enabled) {
-        if (server.masterhost == NULL) {
+        if (iAmMaster()) {
```

Comment:
do we also need to change this one to use iAmMaster() in beforeSleep?
It doesn't feel like a problem, but the code is a bit inconsistent, maybe a overlook?
@oranagra please take a look to make sure there is no problem
```
    /* Run a fast expire cycle (the called function will return
     * ASAP if a fast cycle is not needed). */
    if (server.active_expire_enabled && server.masterhost == NULL)
        activeExpireCycle(ACTIVE_EXPIRE_CYCLE_FAST);
```

Reply:
i think you're right, and i don't understand why it's not an issue.
if it's not an issue, it's just because the "fast" cycle doesn't kick in because of its thresholds.
while we're waiting for more feedback, let's fix it (even if just for consistency).

Reply:
> if it's not an issue, it's just because the "fast" cycle doesn't kick in because of its thresholds.

yean, i mean this, it will always fallback to the databasesCron one. (the fact that the added test never fails like this seems to confirm it?)

ok, i am making the PR

Reply:
it can lead to an issue in cluster mode I mentioned in #11783 , and I used another way to fix it, madelyn suggest using `iAmMaster` to replace all the `server.masterhost == NULL`.